src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -551,16 +550,18 @@ private static ShardWorkerConfig toShardWorkerConfig(Readable input, WorkerOptio
     }
     if (!Strings.isNullOrEmpty(options.publicName)) {
       builder.setPublicName(options.publicName);
+    } else {
+      logger.log(SEVERE, "worker's public name should not be empty");
+      System.exit(0);
```

Comment:
This should not exit with a good (0) status. Recommend that you throw a runtime exception here with the failure instead. An outermost exception handler around this call should be responsible for logging.